src/game/client/line2d.cpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Line 2D
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "line2d.h"
+
+bool Clip_Line_2D(ICoord2D *p1, ICoord2D *p2, ICoord2D *c1, ICoord2D *c2, IRegion2D *clip_region)
+{
+    enum OutCodeEnum
+    {
+        CODE_INSIDE = 0, // 0000
+        CODE_LEFT = 1, // 0001
+        CODE_RIGHT = 2, // 0010
+        CODE_BOTTOM = 4, // 0100
+        CODE_TOP = 8, // 1000
+    };
+
+    int x_min = clip_region->lo.x;
+    int x_max = clip_region->hi.x;
+    int y_min = clip_region->lo.y;
+    int y_max = clip_region->hi.y;
+    int x0 = p1->x;
```

Comment:
We use p1 to represent point 1 and then x0 to represent the coordinates of point 1. Stick to a singular indexing either 0 or 1 not both. Suggest changing the p1 to p0 as it will require fewer changes.

Reply:
Toms told me to name things this way.

Reply:
yeah this is per the algorithm https://en.wikipedia.org/wiki/Cohen%E2%80%93Sutherland_algorithm